lambdas/check-mobile-app-vc-receipt/src/main/java/uk/gov/di/ipv/core/checkmobileappvcreceipt/CheckMobileAppVcReceiptHandler.java

```diff
@@ -165,13 +169,20 @@ private JourneyResponse getJourneyResponse(CheckMobileAppVcReceiptRequest reques
         LogHelper.attachComponentId(configService);
 
         // Retrieve and validate cri response and vc
-        var criResponse = criResponseService.getCriResponseItem(userId, Cri.DCMAW_ASYNC);
+        var criResponse = criResponseService.getCriResponseItem(userId, DCMAW_ASYNC);
         if (criResponse == null) {
             throw new InvalidCriResponseException(ErrorResponse.CRI_RESPONSE_ITEM_NOT_FOUND);
         }
 
-        var vc = verifiableCredentialService.getVc(userId, Cri.DCMAW_ASYNC.getId());
-        if (CriResponseService.STATUS_PENDING.equals(criResponse.getStatus()) && vc == null) {
+        var dcmawAsyncVc =
```

Comment:
@MikeCollingwood Would you mind reviewing this change? I've made the assumption that we can just return a 404 if the VC isn't found, regardless of the status of the CRI response, but I'm missing all of the context so am quite probably wrong.
I think that the previous version would have blown up if the status was anytthing other than pending and there was no VC found, as the use of `List.of()` on line 198 doesn't allow nulls as input.

Reply:
We return a not found if the VC isn't found and the CriResponseItem status is pending.

The other status checks below should ensure that everything past them will have a non-null vc. I.e. a check on each status directly leads to a return except STATUS_PENDING where the vc is also checked, so the only case that could occur after ln184/194 is STATUS_PENDING and non-null vc. 

This change you've made would work if you put the checks:
```java
        if (CriResponseService.STATUS_ERROR.equals(criResponse.getStatus())) {
            return JOURNEY_ERROR;
        }
        if (CriResponseService.STATUS_ABANDON.equals(criResponse.getStatus())) {
            return JOURNEY_ABANDON;
        }
```
above the check for dcmawAsyncVc.isEmpty()

Reply:
Ah ok, so the only status possible is PENDING, ABANDON or ERROR. Will update, thanks Mike